### DISCUSSION TITLE
how to get query params in tenstack router?

### DISCUSSION THREAD

**Discussion by Mhdph** [TanStack/router#397] [Q&A]
hi how to get query params in tenstack router?

**Comment by Mhdph** [comment]
i dont know its the best way to get the params but i found that way 
 
`const routeConfig = createRouteConfig().createChildren((createRoute) => [
  createRoute({
    path: "/",
    component: Index,
  }),
  createRoute({
    path: "about/:id",
    component: About,
  }),
]);` 
 
`const {
    params: { id },
  } = router.useMatch("/about/:id");`

**Reply by djcuze** [reply]
Hey Mhdph, 

You should be able to access the queryParams like this: 

`const { search } = router.useMatch("/about/:id")` 

and then you'll have access to them from the `search` object. 

You _may_ also need to install `zod` to validate the queryParams like this: 

```tsx
import {z} from "zod"

const routeConfig =
  createRouteConfig({})
    .createChildren(createRoute => [
      createRoute({
        path: "/",
        component: ShoppingCart,
        validateSearch: z.object({
          orderCount: z.number().optional(),
          cartId: z.string().optional(),
        })
      })
    ])

createReactRouter({routeConfig})
```
